lib/engine/game/g_1854/step/route.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/route'
+
+module Engine
+  module Game
+    module G1854
+      module Step
+        class Route < Engine::Step::Route
+          def process_run_routes(action)
+            super
+            return if @round.routes.empty?
+
+            # Close when a train is actually run, as opposed to just reaching the dividends step.
+            # This event is not in the generic abilities, so create one that is the same as in the entities
+            # and rely on the rest of the code to match them up.
+            @game.close_companies_on_event!(action.entity, 'ran_train')
```

Comment:
I don't see how this differs from the default behaviour of triggering this event in the dividends step. That has the same check for `@round.routes.empty?`.

Reply:
The first time the corp operates, you get to the dividend step (but can't pay out), have no trains, and the company was closing early.  This closes only when the corp has actually run a train.

Reply:
The line in `Step::Dividend.process_dividend` is:
`@game.close_companies_on_event!(entity, 'ran_train') unless @round.routes.empty?`
So the company shouldn't be closing unless there is a route. How do you have a route if the corporation doesn't yet have any trains?

Reply:
Not sure what was happening before, but that was my recollection.  I tested it with this step deleted and it does work as advertised.  I'll delete this file.